transmogrifier/sources/xml/datacite.py

```diff
@@ -13,100 +14,34 @@
 class Datacite(XMLTransformer):
     """Datacite transformer."""
 
-    def get_optional_fields(self, xml: Tag) -> dict | None:
+    def get_optional_fields(self, source_record: Tag) -> dict | None:
         """
         Retrieve optional TIMDEX fields from a Datacite XML record.
 
         Overrides metaclass get_optional_fields() method.
 
         Args:
-            xml: A BeautifulSoup Tag representing a single Datacite record in
+            source_record: A BeautifulSoup Tag representing a single Datacite record in
                 oai_datacite XML.
         """
         fields: dict = {}
-        source_record_id = self.get_source_record_id(xml)
+        source_record_id = self.get_source_record_id(source_record)
 
         # alternate_titles
-        for alternate_title in [
-            t for t in xml.find_all("title", string=True) if t.get("titleType")
-        ]:
-            fields.setdefault("alternate_titles", []).append(
-                timdex.AlternateTitle(
-                    value=alternate_title.string,
-                    kind=alternate_title["titleType"],
-                )
-            )
-        # If the record has more than one main title, add extras to alternate_titles
-        for index, title in enumerate(self.get_main_titles(xml)):
-            if index > 0:
-                fields.setdefault("alternate_titles", []).append(
-                    timdex.AlternateTitle(value=title)
-                )
+        fields["alternate_titles"] = self.get_alternate_titles(source_record) or None
 
         # content_type
-        if resource_type := xml.metadata.find("resourceType"):
-            if resource_type.string:
-                fields["notes"] = [
-                    timdex.Note(
-                        value=[resource_type.string], kind="Datacite resource type"
-                    )
-                ]
-            if content_type := resource_type.get("resourceTypeGeneral"):
-                if self.valid_content_types([content_type]):
-                    fields["content_type"] = [content_type]
-                else:
-                    message = f'Record skipped based on content type: "{content_type}"'
-                    raise SkippedRecordEvent(message, source_record_id)
-        else:
-            logger.warning(
-                "Datacite record %s missing required Datacite field resourceType",
-                source_record_id,
-            )
+        fields["content_type"] = (
+            self.get_content_type(source_record, source_record_id) or None
+        )
```

Comment:
Interesting, I see the logic in that but I feel that would be harder to pull out if (more likely when) it's deemed no longer necessary. A find-replace for ` or None` vs. removing:
```
 `if not xxxxx`: 
      return None
```
But curious on @jonavellecuerdo 's thoughts, I can be persuaded

Reply:
I feel like it'd be a small change in any given field method, but would have a larger effect higher up on the field method calling context where they would be totally uniform.

Taking this one as an example:
```python
@classmethod
def get_content_type(cls, source_record: Tag, source_record_id: str) -> list[str]:
    content_types = []
    if resource_type := source_record.metadata.find("resourceType"):
        if content_type := resource_type.get("resourceTypeGeneral"):
            if cls.valid_content_types([content_type]):
                content_types.append(str(content_type))
            else:
                message = f'Record skipped based on content type: "{content_type}"'
                raise SkippedRecordEvent(message, source_record_id)
    else:
        logger.warning(
            "Datacite record %s missing required Datacite field resourceType",
            source_record_id,
        )
    return content_types or None  #  <-------- might be just adding "or None" here?
```

Granted it would vary slightly for each, I also think the friction of thinking deeply about the return types and scenarios would also have benefits.  Feels like these methods should be _the_ value -- whatever it is -- that gets set on the `TIMDEXRecord`.  Now how we handle defualts there... that's another story, but at least we'd have more uniform methods to work with.

Reply:
Ugh, you're totally right that's an easy shift, I withdraw my objection but still curious if @jonavellecuerdo has concerns

Reply:
No concerns. I like the idea of the field methods returning "_the_ value -- whatever it is -- that gets set on the `TIMDEXRecord`". I also think that would align more closely with [the type hints set for the fields in the `TimdexRecord` model](https://github.com/MITLibraries/transmogrifier/blob/main/transmogrifier/models.py#L159-L215).